### PR TITLE
Make Projectiles Only Hit a Variety of Station Objects Unless Clicked on

### PR DESCRIPTION
**Pull request by Cojoke-dot**
## About the PR

Makes a variety of objects around the station not be able to be hit by projectiles unless targeted by the player. The objects include:
- Computers
- Circuit imprinter
- Fax
- Reagent grinder
- Portable Generators
- Electrolyzer
- Centrifuge
- Disposal chutes
- Booze, Soda, and Chem Dispensers
- Microwave
- Gas Condenser
- Vaccine Machines
- Chest Drawer
- Large Instruments
- Hydroponics Trays
- Small Wooden Fences/Gates
- Wireless Cameras
- Artifact Analyzer 

Fixes the collision layers with tall fences( bullets and lasers did not hit them and I'm too lazy to make another pr to fix it)
## Why / Balance

A lot of objects around the map feel like they should be able to be shot over but also should have the option of being destroyed by guns(mostly the stuff that can go on tables). I put some of the shorter machines and all of the instruments in here too.
## Technical details

Adds the RequireProjectileTarget component to the objects listed. If an object did not have collisions with bullets before, the BulletImpassable Layer was added with the component.

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- tweak: A variety of objects around the station now need to be clicked on in order for bullets to hit them.

### CODE REVIEW

File: Content.Shared/Damage/Components/RequireProjectileTargetComponent.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+/// <summary>
+/// Prevent the object from getting hit by projetiles unless you target the object.
+/// </summary>
+[RegisterComponent, NetworkedComponent, AutoGenerateComponentState]
+[Access(typeof(RequireProjectileTargetSystem))]
+public sealed partial class RequireProjectileTargetComponent : Component
+{
+    [DataField, AutoNetworkedField]
+    public bool Active = true;
+}
```
Comment (metalgearsloth) [inline Content.Shared/Damage/Components/RequireProjectileTargetComponent.cs:14]: Didn't we just get a PR with this?

**Comment by Cojoke-dot** [comment]
yeah, just needed the merge, thats all gone now

**Comment by Cojoke-dot** [comment]
Adding the hydroponics tray and wood fences

File: Resources/Prototypes/Entities/Structures/Storage/filing_cabinets.yml [hunk 2]
```diff
@@ -67,6 +67,7 @@
     node: filingCabinet
   - type: StaticPrice
     price: 20
+  - type: RequireProjectileTarget
```
Comment (Lamrr) [inline Resources/Prototypes/Entities/Structures/Storage/filing_cabinets.yml:70]: Filing cabinets, or at least the tall cabinets, should probably be able to be used as cover.

**Review by AJCM-git — CHANGES_REQUESTED** [review]
Add em to the grinders and rechargers as well (a good way to get a sense of what may need this is searching for `drawdepth: SmallObjects ` that can be anchored and using some logic of "can you hide behind this?"). Also remember to double check the children of your prototypes to make sure nothing weird is going on.

A better way to do this btw, would be adding the component to a new SmallConstructibleMachine prototype and make those that have ConstructibleMachine inherit that where it makes sense, same logic applies to the other base prototypes (unless all of their children *are* ""small"")

**Comment by Cojoke-dot** [comment]
> Add em to the grinders...
Reagent grinders?

**Comment by AJCM-git** [comment]
> > Add em to the grinders...
> 
> Reagent grinders?

yeah

File: Resources/Prototypes/Entities/Structures/Walls/fence_wood.yml
Comment (AJCM-git) [inline Resources/Prototypes/Entities/Structures/Walls/fence_wood.yml:1]: why change to machine layer?

**Review by AJCM-git — CHANGES_REQUESTED** [review]
you havent answered my last comment

**Comment by Cojoke-dot** [comment]
> you havent answered my last comment

oh... uh... MachineLayer has the layers that were on the fence before + BulletImpassable and Opaque. Is there a different way I should do this?

**Comment by AJCM-git** [comment]
> > you havent answered my last comment
> 
> oh... uh... MachineLayer has the layers that were on the fence before + BulletImpassable and Opaque. Is there a different way I should do this?

MachineLayer is given with the assumption that something is a machine, that way if someone wants to change how the layer works they dont have to double check that everything it affects is indeed a machine, so either just add those or add a new layer

**Comment by Cojoke-dot** [comment]
ok, got it, I'll change that

**Review by AJCM-git — APPROVED** [review]

**Pull request merged**